draft-ietf-ppm-dap.md

```diff
@@ -1391,124 +1320,227 @@ Otherwise, `out` is the pair `(prep_state, prep_msg)`.
+    case finished: Empty;
+    case failed: ReportShareError;
+  }
+} PrepareStep;
+
+struct {
+  u16 round;
+  opaque agg_param<0..2^32-1>;
+  PartialBatchSelector part_batch_selector;
   PrepareStep prepare_steps<1..2^32-1>;
-} AggregateContinueResp;
+} AggregationJob;
 ~~~
 
-The order of AggregateContinueResp.prepare_steps MUST match that of the
-PrepareStep values in `AggregateContinueReq.prepare_steps`. The helper's
-response to the leader is an HTTP status code 200 OK whose body is the
-AggregateContinueResp and media type is
-"application/dap-aggregate-continue-resp". The helper then awaits the next
-message from the leader.
+An `AggregationJob` consists of:
 
-[[OPEN ISSUE: consider relaxing this ordering constraint. See issue#217.]]
+* The current round of the VDAF preparation protocol.
+
+* The opaque, VDAF-specific aggregation parameter provided during the collection
+  flow (`agg_param`),
+
+  [[OPEN ISSUE: Check that this handling of `agg_param` is appropriate when the
+  definition of Poplar is done.]]
+
+* Information used by the Aggregators to determine how to aggregate each report:
+
+    * For fixed_size tasks, the Leader specifies a "batch ID" that determines
+      the batch to which each report for this aggregation job belongs.
+
+      [OPEN ISSUE: For fixed_size tasks, the Leader is in complete control over
+      which batch a report is included in. For time_interval tasks, the Client
+      has some control, since the timestamp determines which batch window it
+      falls in. Is this desirable from a privacy perspective? If not, it might
+      be simpler to drop the timestamp altogether and have the agg init request
+      specify the batch window instead.]
+
+  The indicated query type MUST match the task's query type. Otherwise, the
+  Helper MUST abort with error "queryMismatch".
+
+* The sequence of report shares to aggregate. The `encrypted_input_share` field
+  of the report share is the `HpkeCiphertext` whose index in
+  `Report.encrypted_input_shares` is equal to the index of the aggregator in the
+  task's `aggregator_endpoints` to which the AggregateInitializeReq is being
+  sent.
+
+#### Required methods
+
+##### PUT `/tasks/{task-id}/aggregation_jobs` {#aggregation-job-put}
+
+The Leader begins an aggregation job by choosing a set of candidate reports that
+pertain to the same DAP task and compiling them into an `AggregationJob` (see
+{{aggregation-job-representation}}. The Leader can prepare many reports in
+parallel by concurrently creating multiple aggregation jobs. Creating
+aggregation jobs is how the Leader transmits report shares to the helper, so
+each `PrepareStep.prepare_step_state` MUST be set to `PrepareStepState.start`
+and the `PrepareStep` MUST contain a `ReportShare`. `AggregationJob.round` MUST
+be 0.
+
+If successful, the Helper's responds with HTTP status 303 See Other and a
+Location header containing a URI for the aggregation job. The structure of this
```

Comment:
I'm not certain if explicitly spelling out HTTP 303 See Other and a Location header is a good idea. I think we could just say something like "the Helper redirects to a URI for the aggregation job", and rely on existing HTTP specifications for things like the status code and the headers and so on. On the other hand, if we specify this explicitly as I have here, then there's less ambiguity and less code to write for DAP implementations (with this text, they can simply ignore any 3xx status that isn't 303 instead of having to reason about the nuanced difference between 301 and 303). So I lean towards being specific.